crates/biome_service/src/file_handlers/json.rs

```diff
@@ -179,16 +179,28 @@ fn parse(
     let overrides = settings.map(|s| &s.override_settings);
     let optional_json_file_source = file_source.to_json_file_source();
     let options = JsonParserOptions {
-        allow_comments: parser.map(|p| p.allow_comments).unwrap_or_default()
-            || optional_json_file_source.map_or(false, |x| x.allow_comments()),
-        allow_trailing_commas: parser.map(|p| p.allow_trailing_commas).unwrap_or_default()
-            || optional_json_file_source.map_or(false, |x| x.allow_trailing_commas()),
+        allow_comments: parser.map(|p| p.allow_comments).unwrap_or_default(),
+        allow_trailing_commas: parser.map(|p| p.allow_trailing_commas).unwrap_or_default(),
     };
-    let options = if let Some(overrides) = overrides {
+
+    let mut options = if let Some(overrides) = overrides {
         overrides.to_override_json_parser_options(biome_path, options)
     } else {
         options
     };
+    let options_file_source = JsonParserOptions {
+        allow_comments: optional_json_file_source.map_or(false, |x| x.allow_comments()),
+        allow_trailing_commas: optional_json_file_source
+            .map_or(false, |x| x.allow_trailing_commas()),
+    };
+    // HACK: prevent overrides from affecting how special json files are parsed
+    if options_file_source.allow_comments {
+        options.allow_comments = true;
+    }
+    if options_file_source.allow_trailing_commas {
+        options.allow_trailing_commas = true;
+    }
```

Comment:
I don't think we should prioritize "file source" (well-known files) specific options over "overrides" options. For example, if a user wants to forbid comments or trailing commas in `tsconfig.json`, they should be able to achieve that.

File source specific options should be the default fallback < then normal options < then overrides options

I think we need to fix this problem in `.to_override_json_parser_options`.